python/ray/data/_internal/arrow_block.py

```diff
@@ -414,37 +415,34 @@ def sum_of_squared_diffs_from_mean(
     def sort_and_partition(
         self, boundaries: List[T], key: "SortKeyT", descending: bool
     ) -> List["Block"]:
-        if len(key) > 1:
-            raise NotImplementedError(
-                "sorting by multiple columns is not supported yet"
-            )
-
         if self._table.num_rows == 0:
             # If the pyarrow table is empty we may not have schema
             # so calling sort_indices() will raise an error.
             return [self._empty_table() for _ in range(len(boundaries) + 1)]
 
         context = DataContext.get_current()
         sort = get_sort_transform(context)
-        col, _ = key[0]
+        columns = [k[0] for k in key]
         table = sort(self._table, key, descending)
         if len(boundaries) == 0:
             return [table]
 
-        partitions = []
         # For each boundary value, count the number of items that are less
         # than it. Since the block is sorted, these counts partition the items
         # such that boundaries[i] <= x < boundaries[i + 1] for each x in
         # partition[i]. If `descending` is true, `boundaries` would also be
         # in descending order and we only need to count the number of items
         # *greater than* the boundary value instead.
```

Comment:
`np.searchsorted` does not support multi-dimensional arrays. This implementation over multiple keys is not vectorised and therefore not as optimal/performant. Would appreciate pointers on how to improve it.

Reply:
it's weird that I find we are already using `to_pylist` and our CI also test against pyarrow 6. Not sure why that can work. I'll confirm. also, in this is indeed needed, we can move the definition to `python/ray/data/_internal/arrow_ops/transform_pyarrow.py`.

Reply:
CI for pyarrow 6 was [failing](https://buildkite.com/ray-project/oss-ci-build-pr/builds/28395#018949ee-1589-4c63-8ffe-e4043f11c4e5). Thanks, moved to `arrow_ops`.